File: src/com/machfour/macros/validation/SchemaViolation.java
package com.machfour.macros.validation;

import com.machfour.macros.data.Column;

import java.util.Map;

public class SchemaViolation extends RuntimeException {
    public SchemaViolation(Column<?> c, ValidationError v) {
        super("Schema error (" + v + ") found in column: " + c.sqlName());
    }

    public SchemaViolation(Map<Column<?>, ValidationError> errorMap) {
        super("Schema errors found:" + errorMap.toString());
    }
}

File: src/main/java/com/machfour/macros/validation/SchemaViolation.kt
package com.machfour.macros.validation

import com.machfour.macros.core.Column

class SchemaViolation : RuntimeException {
    constructor (c: Column<*, *>, v: ValidationError) : super("Schema error (" + v + ") found in column: " + c.sqlName)
    constructor(errorMap: Map<out Column<*, *>, List<ValidationError>>) : super("Schema errors found:$errorMap")
}